app/full_text_search.py

```diff
@@ -109,12 +113,35 @@ def search(repo, query_txt, max_results):
     # enclose_in_double_quotes is used for avoiding query_txt
     # which specifies index field name, contains special symbol, ...
     # (e.g., "repo: repository_name", "test: test", "test AND test").
-    and_query = enclose_in_double_quotes(query_txt) + ' AND (repo: ' + repo + ')'
+    and_query = enclose_in_double_quotes(romanized_query) + (
+        ' AND (repo: ' + repo + ')')
+    and_query_with_kanji = enclose_in_double_quotes(
+        romanized_query_with_kanji) + ' AND (repo: ' + repo + ')'
+
     person_location_index_results = person_location_index.search(
         appengine_search.Query(
             query_string=and_query, options=options))
+
+    person_location_index_results_with_kanji = person_location_index.search(
+        appengine_search.Query(
+            query_string=and_query_with_kanji, options=options)
+    )
+
     index_results = []
-    regexp = make_or_regexp(query_txt)
+    regexp = make_or_regexp(romanized_query)
+    for document in person_location_index_results_with_kanji:
```

Comment:
The code here is very similar to the code at line 145. Can you define a function instead?

In general, repeating the same code multiple times is considered bad. It makes harder to read, and you need to update both codes when you need to modify the code. Please consider defining a function instead before you copy&paste multiple lines of code.